### PR TITLE
cteate base

### PR DESCRIPTION
**Pull request by vitconpc**
function: create base project
https://edu-redmine.framgia.vn/issues/103
ready

### CODE REVIEW

File: app/src/main/java/vn/com/example/soundclound/common/Constant.java [hunk 1]
```diff
@@ -0,0 +1,4 @@
+package vn.com.example.soundclound.common;
+
+public class Constant {
```
Comment (DoanVanToan) [inline app/src/main/java/vn/com/example/soundclound/common/Constant.java]: rename to Constants

File: app/src/main/java/vn/com/example/soundclound/presenter/base/BasePresenter.java [hunk 2]
```diff
@@ -0,0 +1,6 @@
+package vn.com.example.soundclound.presenter.base;
+
+public interface BasePresenter {
+    void onViewDestroy();
```
Comment (DoanVanToan) [inline app/src/main/java/vn/com/example/soundclound/presenter/base/BasePresenter.java]: presenter quan tâm đến onViewDestroy làm gì vậy

File: app/src/main/java/vn/com/example/soundclound/common/ResponseCode.java [hunk 3]
```diff
@@ -0,0 +1,4 @@
+package vn.com.example.soundclound.common;
```
Comment (DoanVanToan) [inline app/src/main/java/vn/com/example/soundclound/common/ResponseCode.java]: đặt trong package data> model

File: app/src/main/java/vn/com/example/soundclound/view/base/activity/BaseActivity.java [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+
+    @Override
+    protected void onDestroy() {
+        super.onDestroy();
+        if (mPresenter != null) {
+            mPresenter.onViewDestroy();
+        }
+    }
+
+}
```
Comment (hoanghuong) [inline app/src/main/java/vn/com/example/soundclound/view/base/activity/BaseActivity.java]: blank line

File: app/src/main/java/vn/com/example/soundclound/view/base/fragment/BaseFragment.java [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+    @Override
+    public View onCreateView(LayoutInflater inflater, @Nullable ViewGroup container, Bundle savedInstanceState) {
+        View rootView = inflater.inflate(getLayoutResource(), container, false);
+        mPresenter = initPresenter();
+        initView(savedInstanceState, rootView);
+        initData(savedInstanceState);
+        return super.onCreateView(inflater, container, savedInstanceState);
+    }
+
+}
```
Comment (hoanghuong) [inline app/src/main/java/vn/com/example/soundclound/view/base/fragment/BaseFragment.java]: blank line

**Pull request merged**